src/main/java/org/opensrp/service/OrganizationService.java

```diff
@@ -358,4 +359,15 @@ public void unassignLocationAndPlan(String planIdentifier) {
 	public long countAllOrganizations() {
 		return organizationRepository.countAllOrganizations();
 	}
+
+	@PreAuthorize("hasRole('ORGANIZATION_VIEW')")
+	@PostFilter("hasPermission(filterObject, 'ORGANIZATION_VIEW')")
+	public List<Organization> getAllOrganizationsByOrganizationIds(List<Long> organizationIds) {
+		return organizationRepository.getOrganizationsByIds(organizationIds);
+	}
+
+	public List<Organization> getOrganizationsByPractitionerIdentifier(String practitionerIdentifier) {
```

Comment:
Please add the following to filter by roles and permissions
```
@PreAuthorize("hasRole('ORGANIZATION_VIEW')")
@PostFilter("hasPermission(filterObject, 'ORGANIZATION_VIEW')")
```

Reply:
Added